pynestml/codegeneration/nest_builder.py

```diff
@@ -146,23 +148,36 @@ def build(self) -> None:
         else:
             shell = False
 
-        # first call cmake with all the arguments
-        try:
-            subprocess.check_call(cmake_cmd, stderr=subprocess.STDOUT, shell=shell,
-                                  cwd=str(os.path.join(target_path)))
-        except subprocess.CalledProcessError as e:
-            raise GeneratedCodeBuildException('Error occurred during \'cmake\'! More detailed error messages can be found in stdout.')
+        stages_exception = {"cmake": f"Error occurred during cmake! More detailed error messages can be found in {error_location}.", "build": f"Error occurred during 'make all'! More detailed error messages can be found in {error_location}.",
+                            "install":  f"Error occurred during 'make install'! More detailed error messages can be found in {error_location}."}
+        current_stage = ""
 
-        # now execute make all
-        try:
-            subprocess.check_call(make_all_cmd, stderr=subprocess.STDOUT, shell=shell,
-                                  cwd=str(os.path.join(target_path)))
-        except subprocess.CalledProcessError as e:
-            raise GeneratedCodeBuildException('Error occurred during \'make all\'! More detailed error messages can be found in stdout.')
+        stdout = self._options["stdout"]
+        stderr = self._options["stderr"]
+
+        if self._options["redirect"]:
+            try:
```

Comment:
It's probably not anything too catastrophic, but putting two things that can fail independently inside the same ``try`` block might not be best practice. What if stdout is opened successfully but stderr fails to open with an unhandeled exception, and we drop into the iPython prompt with a still-open stdout stream? To avoid having to consider all these edge cases, and because the opening and closing of the streams is now anyway very close to where the streams are being used, can we just use a context manager here? See my earlier comment (https://github.com/nest/nestml/pull/741#discussion_r1101229678).

Reply:
Sorry, I just made that comment, when I was in the train, and it seems that I have not given it enough thoughts.

I think the best, as you said, to keep opening the file locally close to where they are needed. Therefore, I have split the creation of the files into two different `try-catch` blocks.